src/main/java/bio/terra/service/auth/iam/sam/SamIam.java

```diff
@@ -314,6 +315,58 @@ private Map<IamRole, String> syncSnapshotResourcePoliciesInner(
     return policies;
   }
 
+  @Override
+  public Map<IamRole, List<String>> createSnapshotBuilderRequestResource(
+      AuthenticatedUserRequest userReq, UUID snapshotId, UUID snapshotBuilderRequestId)
+      throws InterruptedException {
+    var initialRoles = Map.of(IamRole.OWNER, List.of(userReq.getEmail()));
+    SamRetry.retry(
+        configurationService,
+        () ->
+            createSnapshotBuilderRequestResourceInner(
+                userReq, snapshotId, snapshotBuilderRequestId, initialRoles));
+    return initialRoles;
+  }
+
+  private void createSnapshotBuilderRequestResourceInner(
+      AuthenticatedUserRequest userReq,
+      UUID snapshotId,
+      UUID snapshotBuilderRequestId,
+      Map<IamRole, List<String>> initialRoles)
+      throws ApiException {
+    ResourcesApi samResourceApi = samApiService.resourcesApi(userReq.getToken());
+    CreateResourceRequestV2 req =
+        createSnapshotBuilderRequestResourceRequest(
+            userReq, snapshotId, snapshotBuilderRequestId, initialRoles);
+    samResourceApi.createResourceV2(IamResourceType.SNAPSHOT_BUILDER_REQUEST.toString(), req);
+  }
+
+  @VisibleForTesting
+  CreateResourceRequestV2 createSnapshotBuilderRequestResourceRequest(
```

Comment:
Do we want to make this `private` scoped?

Reply:
I had it visible for testing, but I don't think I'm actually testing this directly so it should be fine.